tests/test_pipeline.py

```diff
@@ -275,6 +316,90 @@ def test_gridsearchcv(self) -> None:
             for k, value in param_grid.items():
                 self.assertIn(grid_search_cv.best_params_[k], value)
 
+    def test_caching(self) -> None:
+        """Test if the caching gives the same results and is faster on the second run."""
+
+        molecule_net_logd_df = pd.read_csv(
+            "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/Lipophilicity.csv"
```

Comment:
No. Please don't make our unittest depend on some external resource requiring an internet connection. The unit test should be self-contained within the repo. Instead you should add an CSV file in the test data directory.

Reply:
Done